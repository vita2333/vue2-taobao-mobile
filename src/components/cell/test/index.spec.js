import Cell from '..'
import CellGroup from '../../cell-group'
import { mount } from '../../../../tests'

test('click event', () => {
  const click = jest.fn()
  const wrapper = mount(Cell, {
    context: {
      on: {
        click,
      },
    },
  })

  wrapper.trigger('click')
  expect(click).toHaveBeenCalled()
})

test('render slot', () => {
  const wrapper = mount({
    template: `
      <cell>
        <template v-slot:title>Custom Title</template>
        <template v-slot:label>Custom Label</template>
      </cell>
    `,
    components: {
      Cell,
    },
  })

  expect(wrapper).toMatchSnapshot()
})

test('CellGroup title slot', () => {
  const wrapper = mount(CellGroup, {
    scopedSlots: {
      title: () => 'CustomTitle',
    },
  })

  expect(wrapper).toMatchSnapshot()
})
