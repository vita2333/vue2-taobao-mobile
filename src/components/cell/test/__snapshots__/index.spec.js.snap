// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellGroup title slot 1`] = `
<div class="vita-cell-group">
  <div class="vita-cell-group__title">CustomTitle</div>
  <div></div>
</div>
`;

exports[`render slot 1`] = `
<div class="vita-cell">
  <div class="vita-cell__title">Custom Title<div class="vita-cell__label">Custom Label</div>
  </div>
  <div class="vita-cell__value"><span></span></div>
</div>
`;
